src/audioWorklet/soundFileProcessor.js

```diff
@@ -2,12 +2,29 @@
 const processorNames = preval.require('./processorNames');
 
 class SoundFileProcessor extends AudioWorkletProcessor {
+  constructor() {
+    super();
+
+    this.paused = false;
+
+    this.port.onmessage = (event) => {
+      const data = event.data;
+      if (data.name === 'play') {
+        this.paused = false;
+      } else if (data.name === 'pause') {
+        this.paused = true;
+      }
+    };
+  }
+
   process(inputs) {
     const input = inputs[0];
     const inputChannel = input[0];
     const position = inputChannel[inputChannel.length - 1] || 0;
 
-    this.port.postMessage({ name: 'position', position: position });
+    if (!this.paused) {
+      this.port.postMessage({ name: 'position', position: position });
```

Comment:
it seems we can't rely on `this.paused` having updated in Safari. Maybe this logic needs to be in the SoundFile?

Reply:
sure, it's easy enough to move this logic to the SoundFile class